### ISSUE TITLE
DTD generation fails on <dataRef> as content of <content>

### ISSUE THREAD

**Issue by sydb** [TEIC/Stylesheets#157]
As mentioned in [a Guildelines ticket](https://github.com/TEIC/TEI/issues/1464), the following content model causes invalid DTDs on output.
~~~~~xml
  <content>
    <dataRef key="teidata.name"/>
  </content>
~~~~~

In the pre-PureODD days we used to use
~~~~~xml
  <content>
    <macroRef key="data.name"/>
  </content>
~~~~~
for this. The Stylesheets somehow knew that when used inside `<content>`, a reference to a datatype should be resolved differently than when used inside `<attDef>`. But they don’t seem to know this for the new `<dataRef>`. In this particular case it should be resolved to `(#PCDATA)` instead of `%teidata.name;` (which resolves to `' CDATA '`).

**This prevents P5 from building properly**, and therefore is a serious bug that should be addressed quickly, IMHO. I hope to try to look around at it a bit later this week, but have to turn my attention to other things now, at least for awhile. Furthermore, I may not have much success.

**Comment by sydb** [comment]
Fixed, I hope, in 0d226da. If Mr. Jenkins is happy, I’ll close this ticket.

**Comment by sydb** [comment]
Mr. Jenkins is a smilin’
